### PR TITLE
Review all uses of Any and refer to relevant issues

### PR DESCRIPTION
**Pull request by happz**
Some `Any` are useful, for some I filed either an issue or an alias to imrpvoe the situation for future devels.

### CODE REVIEW

File: tmt/steps/provision/testcloud.py [hunk 1]
```diff
@@ -334,7 +338,7 @@ def clean_images(cls, clean: Any, dry: Any) -> bool:
                 try:
                     os.remove(image)
                 except OSError:
-                    clean.error(f"Failed to remove '{image}'.", shift=2)
```
Comment (lukaszachy) [inline tmt/steps/provision/testcloud.py:337]: Good catch.

**Review by psss — CHANGES_REQUESTED** [review]
Just one tiny change.

File: tmt/export.py [hunk 2]
```diff
@@ -354,7 +364,7 @@ def find_polarion_case_ids(
 def get_polarion_case(
         data: Dict[str, Optional[str]],
         preferred_project: Optional[str] = None,
-        polarion_case_id: Optional[str] = None) -> Any:
+        polarion_case_id: Optional[str] = None) -> PolarionTestCase:
```
[the remark this answers is not in the document]
Comment (happz) [reply tmt/export.py:367]: Correct, it's `Optional`.

[the remark this answers is not in the document]
Comment (psss) [reply tmt/export.py:367]: Ok updated.

**Pull request merged**